apps/cowswap-frontend/src/modules/ordersTable/pure/OrdersTableContainer/OrderRow/index.tsx

```diff
@@ -225,22 +231,81 @@ export function OrderRow({
       )
     }
 
-    // For TWAP parent orders, check child states
+    // For TWAP parent orders
     if (isTwapTable && !isChild && childOrders) {
-      // Check if all child orders are filled (100%)
-      const allChildrenFilled = childOrders.every((childOrder) => {
-        const isFullyFilled =
-          childOrder.status === OrderStatus.FULFILLED &&
-          (percentIsAlmostHundred(childOrder.executionData.filledPercentDisplay) ||
-            Number(childOrder.executionData.filledPercentDisplay) >= 99.99)
-        console.debug('Child order fill status:', {
-          orderId: childOrder.id,
-          status: childOrder.status,
-          fillPercent: childOrder.executionData.filledPercentDisplay,
-          isFullyFilled,
-        })
-        return isFullyFilled
-      })
+      // First priority: Check for warnings - MUST check child orders first
+      const warningChild = childOrders.find(
+        (childOrder) =>
+          childOrder.status !== OrderStatus.FULFILLED &&
+          (childOrder.status === OrderStatus.SCHEDULED || childOrder.status === OrderStatus.PENDING) &&
+          (getOrderParams(chainId, balancesAndAllowances, childOrder)?.hasEnoughBalance === false ||
```

Comment:
getOrderParams is called twice here, should be cached to a const

Reply:
Should be addressed now.